server/elm/src/DateSelector/Selector.elm

```diff
@@ -0,0 +1,418 @@
+                            Date.monthToNumber month
+                    in
+                    option
+                        [ value <| String.fromInt monthNumber
+                        , selected <| Date.monthNumber selectedDate == monthNumber
+                        ]
+                        [ text <| translate language <| Translate.ResolveMonth False month ]
+                )
+                months
+    in
+    select
+        [ on "input" <|
+            Json.Decode.map
+                (\s ->
+                    let
+                        selectedMonths =
+                            String.toInt s
+                                |> -- We'll never get here, as we place only
+                                   -- valid numeric values at select options.
+                                   Maybe.withDefault 1
+                    in
+                    numberToMonth selectedMonths
+                        |> dateWithMonth selectedDate
+                )
+                (Json.Decode.at [ "target", "value" ] Json.Decode.string)
+        ]
+        options
+
+
+viewMonthSelectListDisabled : Html a
+viewMonthSelectListDisabled =
+    select [] [ option [ value "" ] [ text "" ] ]
+
+
+groupsOf : Int -> List a -> List (List a)
+groupsOf n list =
+    if List.isEmpty list then
+        []
+
+    else
+        List.take n list :: groupsOf n (List.drop n list)
+
+
+monthDates : Int -> Month -> List Date
+monthDates y m =
+    let
+        start =
+            Date.floor Monday <| Date.fromCalendarDate y m 1
+    in
+    Date.range Day 1 start <| Date.add Days 42 start
+
+
+dateWithYear : Date -> Int -> Date
+dateWithYear date y =
+    let
+        m =
+            month date
+
+        d =
+            day date
+    in
+    if m == Feb && d == 29 && not (isLeapYear y) then
+        Date.fromCalendarDate y Feb 28
+
+    else
+        Date.fromCalendarDate y m d
+
+
+dateWithMonth : Date -> Month -> Date
+dateWithMonth date m =
+    let
+        y =
+            year date
+
+        d =
+            day date
+    in
+    Date.fromCalendarDate y m <| Basics.min d (daysInMonth y m)
+
+
+isLeapYear : Int -> Bool
+isLeapYear y =
+    modBy 4 y == 0 && modBy 100 y /= 0 || modBy 400 y == 0
+
+
+daysInMonth : Int -> Month -> Int
+daysInMonth y m =
+    case m of
+        Jan ->
+            31
+
+        Feb ->
+            if isLeapYear y then
+                29
+
+            else
+                28
+
+        Mar ->
+            31
+
+        Apr ->
+            30
+
+        May ->
+            31
+
+        Jun ->
+            30
+
+        Jul ->
+            31
+
+        Aug ->
+            31
+
+        Sep ->
+            30
+
+        Oct ->
+            31
+
+        Nov ->
+            30
+
+        Dec ->
+            31
+
+
+
+-- view
+
+
+type State
+    = Normal
+    | Dimmed
+    | Disabled
+    | Selected
+
+
+isSelectable : State -> Bool
+isSelectable state =
+    state == Normal || state == Dimmed
```

Comment:
It returns True for both Normal and Dimmed states, which might not be intended behavior.

Reply:
For DateSelector is a Utility we "borrowed" the code long time ago and added minor adjustments to make it look per our need.
We don't get into it's logic - been using it for few years now, and had no complaints so far.